src/main/java/org/jenkinsci/test/acceptance/selenium/Scroller.java

```diff
@@ -73,13 +73,23 @@ public void beforeClickOn(WebElement element, WebDriver driver) {
     }
 
     @Override
-    public void beforeChangeValueOf(WebElement element, WebDriver driver) {
+    public void beforeChangeValueOf(WebElement element, WebDriver driver, CharSequence[] keysToSend) {
         scrollIntoView(element, driver);
     }
 
     private void scrollIntoView(WebElement e, WebDriver driver) {
         int eYCoord = e.getLocation().getY();
         int eXCoord = e.getLocation().getX();
-        ((JavascriptExecutor)driver).executeScript(scrollJs, eYCoord, eXCoord);
+        String id = e.getAttribute("id");
+
+        WebElement response = (WebElement) ((JavascriptExecutor)driver).executeScript(scrollJs, eYCoord, eXCoord, id);
+        if(response!=null && response.getAttribute("style").contains("bottom: 0px")){
```

Comment:
Since we have to add custom handling for some elements to our scrolling logic, I suggest to either keep it either fully in javascript land or the java land. I speculate the reason for the split is that sleeping in JS is tricky. Here are some suggestion I consider cleaner:

- Return from JS a clear indicator we have hit a specific condition to handle (so java code do not have to inspect the CSS attached).
- Or provided this is a temporary condition, return `true` if we scrolled and `false` if we hit the special condition so the java code will sleep and _retry_ the scrolling logic until it succeeds.

Reply:
I moved it into JS which return true if it is successful scrolled and false if we hit the footer overlap condition. In this case there is retry until it succeeds with timeout 5 seconds.